components/scim/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -1150,4 +1224,73 @@ private void provisionSCIMOperation(int provisioningMethod, SCIMObject provision
             throw new CharonException("Error in initializing provisioning handler", e);
         }
     }
+
+    private void provision(ProvisioningOperation provisioningMethod, SCIMObject provisioningObject) throws CharonException {
+
+        String domainName = UserCoreUtil.getDomainName(carbonUM.getRealmConfiguration());
+        try {
+
+            Map<org.wso2.carbon.identity.application.common.model.ClaimMapping, List<String>> outboundAttributes = new HashMap<org.wso2.carbon.identity.application.common.model.ClaimMapping, List<String>>();
+
+            if (provisioningObject.getAttribute("password") != null) {
+                outboundAttributes.put(org.wso2.carbon.identity.application.common.model.ClaimMapping.build(
+                                IdentityProvisioningConstants.PASSWORD_CLAIM_URI, null, null, false),
+                        Arrays.asList(new String[]{((SimpleAttribute) provisioningObject.getAttribute("password"))
+                                .getStringValue()}));
+            }
+
+            if (provisioningObject.getAttribute(USER_NAME_STRING) != null) {
```

Comment:
provisioningObject is passed to 'provision' method from 'patchUser' method. There it use the parameter newUser as the 'provisionigObject'. That method was called in charon 'UserResourceEndpoind' and there we validate this object, in validating we check this for null as well. Do we need to do that check here again? It will be redundant, isn't it?